src/components/OfferSwitch/OfferSwitch.tsx

```diff
@@ -0,0 +1,61 @@
+import classNames from 'classnames';
+import { useTranslation } from 'react-i18next';
+
+import styles from './OfferSwitch.module.scss';
+
+import type { Offer } from '#types/checkout';
+import Checkbox from '#components/Checkbox/Checkbox';
+import { formatLocalizedDate, formatPrice } from '#src/utils/formatting';
+import { useAccountStore } from '#src/stores/AccountStore';
+
+interface OfferSwitchProps {
+  isCurrentOffer: boolean;
+  pendingDowngradeOfferId: string;
+  offer: Offer;
+  selected: {
+    value: boolean;
+    set: React.Dispatch<React.SetStateAction<string | null>>;
+  };
```

Comment:
This is a bit unconventional because you now require the `set` handler to use `React.Dispatch`.

I suggest making it abstract:

```suggestion
  selected: boolean;
  onChange: () => void;
```